### PR TITLE
OD-468 [Feature] Set default sorting orders to the data source list

### PR DESCRIPTION
**Pull request by AndrRyaz**
@sofiiakvasnevska 
## Issue
OD-468 https://weboo.atlassian.net/browse/OD-468
## Description
Added new logic for sorting icons in data source
## Screenshots/screencasts
http://somup.com/crf3IAFbsV
## Backward compatibility
This change is fully backward compatible.
## Reviewers
@upplabs-alex-levchenko @ivandevupp @YaroslavOvdii

### CODE REVIEW

**Comment by tonytlwu** [comment]
See Jira ticket for comment on the screencast.

**Comment by AndrRyaz** [comment]
@tonytlwu here is a fix video http://somup.com/crf0lhFqGa

File: css/interface.css [hunk 1]
```diff
@@ -18,6 +18,16 @@ body {
   align-items: center;
 }
 
+.desc.sorted::after {
+  font: normal normal normal 14px/1 FontAwesome;
```
Comment (squallstar) [inline css/interface.css:22]: @tonytlwu you good with this?

Comment (tonytlwu) [reply css/interface.css:22]: I don't know yet. There are lots of errors with the screencast video. I'll share my thoughts in Jira.

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Rejected +1

See Jira for bugs found with the screencast.

**Comment by AndrRyaz** [comment]
@tonytlwu here is a new video with fixes http://somup.com/crfUlwFUNx

**Comment by tonytlwu** [comment]
@AndrRyaz In 0:46 of http://somup.com/crfUlwFUNx you clicked on **Last updated** for the first time after switching to a different column. Whenever there is a switch to sort by a different column, the column must start by being sorted:

* Data source: Ascending
* Last updated: Descending
* Deleted: Descending

**Comment by AndrRyaz** [comment]
@tonytlwu i think something goes wrong because i retest it right now and here is a video - http://somup.com/crfvjbFwkI

**Comment by tonytlwu** [comment]
@AndrRyaz In 0:33 of http://somup.com/crfUlwFUNx you switched Last updated to to ASC. You then switched to the Trash view before clicking on Last updated again at 0:46. It seems that although the columns are acting independently of each other, they are not always starting with the right sorting order when you switch away from the column and back again.

You didn't have such a scenario in http://somup.com/crfvjbFwkI so I can't be confident that you have fixed the issue.

**Comment by AndrRyaz** [comment]
@tonytlwu do you mean that i should reset sorting order when switching between tabs (All/Trash)?

**Comment by tonytlwu** [comment]
@AndrRyaz No, the sorting order doesn't need to reset when switching between All & Trash tabs. However, the first time you click on a column to sort, it should always start with the right sorting order. In 0:33 of http://somup.com/crfUlwFUNx you switched Last updated to to ASC. You then switched to the Trash view before clicking on Last updated again at 0:46. Last updated was then sorted using ASC when it should be DESC because it's the first time you switched to that column.

Like in 0:40, this is the first time the table is sorted by Deleted at, so it should have been sorted in DESC, not ASC.

File: js/interface.js [hunk 2]
```diff
@@ -914,28 +908,33 @@ $(window).on('resize', windowResized).trigger('resize');
 $('#app')
   .on('click', '[data-order-date]', function() {
     var $dataSource = $(this);
+    var $defaultOrder = $(this).data('defaultOrder');
```
Comment (tonytlwu) [inline js/interface.js:911]: defaultOrder is not a jQuery object.

```suggestion
    var defaultOrder = $(this).data('defaultOrder');
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Rejected +1

**Comment by tonytlwu** [comment]
@AndrRyaz Please show a new screencast of the latest solution.

File: js/interface.js (hunk 2 above)
Comment (tonytlwu) [reply js/interface.js:912]: Unresolved issue. Don't use `$` prefix when it's not a jQuery object.

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Rejected +1

**Comment by AndrRyaz** [comment]
@tonytlwu fixed

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: Medium
Affected: Data source lists
Verify: Data source lists are rendered and ordered as expected

Squash and merge: Use PR title

**Pull request merged**